docs/src/upgrade/release-3.7.x.asciidoc

```diff
@@ -33,6 +33,9 @@ complete list of all the modifications that are part of this release.
 
 === Upgrading for Providers
 
+==== Renaming None Step to Discard
```

Comment:
i suggest "Deprecated none() step" as a title so that "deprecate" is clearly there in the toc.

Reply:
Will do, thanks!

Reply:
Updated providers section and added user upgrade section